pkg/scheduler/framework/plugins/noderesources/least_allocated.go

```diff
@@ -63,25 +64,58 @@ func (la *LeastAllocated) ScoreExtensions() framework.ScoreExtensions {
 }
 
 // NewLeastAllocated initializes a new plugin and returns it.
-func NewLeastAllocated(_ runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+// resource without the weight defined or explicitly defined as zero will be applied with defined weight, i.e. "1" instead,
+// if the resource is not accountable for the final score, just remove the resource from the plugin args.
+func NewLeastAllocated(laArgs runtime.Object, h framework.FrameworkHandle) (framework.Plugin, error) {
+	args, err := getNodeResourcesLeastAllocatedArgs(laArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	resToWeightMap := make(resourceToWeightMap)
+	if len(args.Resources) == 0 {
+		// If no resources specified, used the default set.
+		resToWeightMap = defaultRequestedRatioResources
+	}
+	for _, resource := range args.Resources {
+		resToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
+		if resource.Weight == 0 {
+			// Apply the default weight.
+			resToWeightMap[v1.ResourceName(resource.Name)] = 1
+		}
+	}
+
 	return &LeastAllocated{
 		handle: h,
 		resourceAllocationScorer: resourceAllocationScorer{
-			LeastAllocatedName,
-			leastResourceScorer,
-			defaultRequestedRatioResources,
+			Name:                LeastAllocatedName,
+			scorer:              leastResourceScorer(resToWeightMap),
+			resourceToWeightMap: resToWeightMap,
 		},
 	}, nil
 }
 
-func leastResourceScorer(requested, allocable resourceToValueMap, includeVolumes bool, requestedVolumes int, allocatableVolumes int) int64 {
-	var nodeScore, weightSum int64
-	for resource, weight := range defaultRequestedRatioResources {
-		resourceScore := leastRequestedScore(requested[resource], allocable[resource])
-		nodeScore += resourceScore * weight
-		weightSum += weight
+func getNodeResourcesLeastAllocatedArgs(obj runtime.Object) (config.NodeResourcesLeastAllocatedArgs, error) {
+	if obj == nil {
```

Comment:
You can now drop this conditional. This is handled in framework.New

Reply:
done